libvcx/src/settings.rs

```diff
@@ -438,26 +383,16 @@ impl ::std::string::ToString for ProtocolTypes {
     }
 }
 
-pub fn get_protocol_type() -> ProtocolTypes {
-    ::std::env::var("CONFIG_PROTOCOL_TYPE")
-        .unwrap_or_else(|_e| {
-            warn!("Env variable CONFIG_PROTOCOL_TYPE was not set.");
-            get_config_value(CONFIG_PROTOCOL_TYPE).unwrap_or_else(|_e| {
-                error!("Config CONFIG_PROTOCOL_TYPE was not set. Will use default value of 3.0");
-                "3.0".into()
-            })
-        })
-        .into()
-}
-
 pub fn clear_config() {
     trace!("clear_config >>>");
     let mut config = SETTINGS.write().unwrap();
     config.clear();
+    agency_settings::clear_config_agency()
```

Comment:
One of the reasons all settings should share a module, IMO

Reply:
Disagree, this "redirection" is to simply preserve `aries-vcx` functionality as a whole, but removing dependencies on `aries-vcx` globals is precondition to create independent modules.

Reply:
What do you mean by redirection? I think that on the contrary, `libvcx`, as a "user" of `agency_comm`, has the responsibility of configuring it as necessary. As I see it, `agency_settings`is a subset of `libvcx` settings, not separate from it.

Reply:
By redirection I meant that call into the `src/settings.rs` further calls `agency_settings.rs`.

But yeah, it could be removed and code in all places would have to make calls into `agency_settings.rs` manually when needed. It was more convenient to implement this way, with "the redirection", but I also don't see this as ultimate state. I think we could eliminate `agency_settings.rs` completely in the next steps.